Intersect (Core)/Enums/GameObjectTypeExtensions.cs

```diff
@@ -54,5 +55,111 @@ public static IDatabaseObject CreateNew(this GameObjectType gameObjectType)
 
             return instance as IDatabaseObject;
         }
+
+        public static int ListIndex(this GameObjectType gameObjectType, Guid id, VariableDataType dataTypeFilter = 0)
+        {
+            var lookup = gameObjectType.GetLookup();
+
+            if (dataTypeFilter == 0)
+            {
+                return lookup.KeyList.OrderBy(pairs => lookup[pairs]?.Name).ToList().IndexOf(id);
+            }
+
+            try
+            {
+                return lookup
+                    .OrderBy(kv => kv.Value?.Name)
+                    .Select(kv => kv.Value)
+                    .Cast<IVariableBase>()
+                    .Where(desc => desc.Type == dataTypeFilter)
+                    .Select(desc => desc.Id)
+                    .ToList()
+                    .IndexOf(id);
+            }
+            catch (InvalidCastException e)
+            {
+                Logging.Log.Error($"Invalid cast! A VariableDataFilter was probably passed for a game object that does not inherit IVariableBase!");
+                return -1;
+            }
+        }
+
+        public static VariableDataType GetVariableType(this GameObjectType gameObjectType, Guid variableDescriptorId)
+        {
+            var lookup = gameObjectType.GetLookup();
+
+            try
+            {
+                return lookup.ValueList
+                    .Cast<IVariableBase>()
+                    .FirstOrDefault(var => var.Id == variableDescriptorId)?.Type ?? 0;
+            }
+            catch (InvalidCastException e)
+            {
+                Logging.Log.Error($"Invalid cast! Tried to get a variable value for a GameObject that is not represented by a class inheriting IVariableBase!");
+                return 0;
+            }
+        }
+
+        public static Guid IdFromList(this GameObjectType gameObjectType, int listIndex, VariableDataType dataTypeFilter = 0)
+        {
+            var lookup = gameObjectType.GetLookup();
+
+            if (listIndex < 0 || listIndex > lookup.KeyList.Count)
```

Comment:
This does not account for the empty lookup case, should be listIndex >= lookup.KeyList.Count